doc/RichTextArea/RichTextArea.md

```diff
@@ -0,0 +1,742 @@
+
+![embedded nodes](embedded-nodes.png)
+
+```java
+public class ExamplesModel extends StyledTextModelViewOnlyBase {
+    /** properties in the model allow for inline controls */
+    private final SimpleStringProperty exampleProperty = new SimpleStringProperty();
+
+    @Override
+    public RichParagraph getParagraph(int index) {
+        switch(index) {
+        case 0: // this model contains a single paragraph
+            RichParagraph.Builder b = RichParagraph.builder();
+            b.addSegment("Input field: ");
+            // creates an embedded control bound to a property within this model
+            b.addInlineNode(() -> {
+               TextField t = new TextField();
+               t.textProperty().bindBidirectional(exampleProperty);
+               return t;
+            });
+            return b.build();
+        }
+        return null;
+    }
+
+    ...
+}
+```
+
+
+
+##### Styling
+
+There are two ways of styling text in RichTextArea: either using inline attributes, or relying on style names in the application style sheet.
+
+It is important to understand the limitation of stylesheet approach as it is only suitable for view-only models because editing of styles by the user is not possible when styles come from the application stylesheet (and might be dynamically changed by the application).  Using stylesheet's names also necessitates the use of **StyleResolver** in order to obtain the actual attribute values when they are needed (for instance, when exporting).  And since the stylesheet might specify totally different styles for different RichTextAreas, the visual rendering and the exported attributes might differ, even when both controls are connected to the same model.
+
+This example illustrates how to style a read-only document created with the SimpleViewOnlyStyledModel and an application stylesheet:
+
+```java
+        SimpleViewOnlyStyledModel m = new SimpleViewOnlyStyledModel();
+        // add text segment using CSS style name (requires a style sheet)
+        m.addSegment("RichTextArea ", null, "HEADER");
+        // add text segment using direct style
+        m.addSegment("Demo", "-fx-font-size:200%;", null);
+        // newline
+        m.nl();
+```
+
+The default model for RichTextArea, the EditableRichTextModel, uses a number of style attributes (found in StyleAttrs class).  These attributes are applicable either to the whole paragraph (BACKGROUND, BULLET, FIRST_LINE_INDENT, ...) or to the individual text segments (BOLD, FONT_FAMILY, etc.).
+
+
+
+#### Export And Import
+
+StyledTextModel provides a common mechanism for importing/exporting styled text into/from the model via the following methods:
+
+- void **export**(TextPos start, TextPos end, StyledOutput out)
+- TextPos **replace**(StyleResolver, TextPos start, TextPos end, String text, boolean allowUndo)
+- TextPos **replace**(StyleResolver, TextPos start, TextPos end, StyledInput in, boolean allowUndo)
```

Comment:
This is confusing and asymmetrical. The `export` method takes a `StyledOutput` as expected. Shouldn't there be a corresponding import method that takes a `StyledInput` instead of reusing the replace method?